starknet_py/serialization/data_serializers/enum_serializer.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import Dict, Generator
+
+from indexed import IndexedOrderedDict
+
+from starknet_py.serialization._context import (
+    DeserializationContext,
+    SerializationContext,
+)
+from starknet_py.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.serialization.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class EnumSerializer(CairoDataSerializer[Dict, TupleDataclass]):
+    """
+    Serializer of enums.
+    Can serialize a dictionary.
+    Deserializes data to a TupleDataclass.
+
+    Example:
+    {"a": 1} => [0, 1]
+    {"b": 100} => [1, 100]
+    """
+
+    serializers: IndexedOrderedDict[str, CairoDataSerializer]
+
+    def deserialize_with_context(
+        self, context: DeserializationContext
+    ) -> TupleDataclass:
+        [variant_index] = context.reader.read(1)
+        variant_name, serializer = self.serializers.items()[variant_index]
```

Comment:
I'm not too sure about using the new dependency. We could just iterate over the items and get the variant index.